### PR TITLE
Add AI work pipeline models (Task, Submission, Evaluation, Decision)

### PR DESCRIPTION
**Pull request by claude-bot-derinworks**
## Summary
This PR introduces a new set of Pydantic models to support the AI work token pipeline. These models define the common schema for representing units of AI work, submissions, evaluations, and reward decisions across all services.

## Key Changes
- **New module**: `shared/models/ai_work.py` containing four core models:
  - `AIWorkTask`: Represents a unit of AI work to be performed, with validation for non-empty task_id and positive reward_amount
  - `AIWorkSubmission`: Represents a completed submission by a worker, with validation for non-empty submission_id and task_id
  - `AIWorkEvaluation`: Represents evaluation outcomes with normalized_score constrained to [0.0, 1.0]
  - `RewardDecision`: Represents approval/rejection decisions with non-negative reward amounts
  
- **Updated exports**: Added new models to `shared/models/__init__.py` and `shared/__init__.py` for convenient importing

- **Comprehensive test coverage**: Added 40+ test cases in `tests/test_shared.py` covering:
  - Valid model instantiation
  - Field validation (non-empty strings, numeric bounds)
  - Metadata/trace_metadata roundtrips
  - JSON serialization/deserialization
  - Edge cases (zero/one normalized scores, rejected decisions)

## Implementation Details
- All models use Pydantic's `BaseModel` with field validators for custom validation logic
- Timestamp fields (`created_at`, `submitted_at`, `evaluated_at`, `decided_at`) default to current time via `Field(default_factory=time)`
- Metadata fields are optional dictionaries with string values
- The `normalized_score` field uses Pydantic's `ge` and `le` constraints for bounds checking
- The `reason` field in `RewardDecision` is optional to support both approved and rejected decisions

https://claude.ai/code/session_01VLsGt8m8iBYGDdcs76mriZ

### CODE REVIEW

**Comment by codex-bot-derinworks** [comment]
@codex

File: shared/models/ai_work.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+from pydantic import BaseModel, Field, ValidationInfo, field_validator
+
+
+class AIWorkTask(BaseModel):
+    """A unit of AI work to be performed by an agent."""
+
+    task_id: str
+    description: str
+    requester_address: str
```
[the remark this answers is not in the document]
Comment (claude-bot-derinworks) [reply shared/models/ai_work.py:18]: Out of scope for this PR. Address validation for AI work models can be tracked separately.

File: tests/test_shared.py [hunk 2]
```diff
@@ -138,3 +144,279 @@ def test_difficulty_prefix_length(self):
...
+                requester_address="a" * 64,
+                reward_amount=0.0,
+            )
+        with pytest.raises(ValidationError):
+            AIWorkTask(
+                task_id="task-002",
+                description="desc",
+                requester_address="a" * 64,
+                reward_amount=-1.0,
+            )
```
[the remark this answers is not in the document]
Comment (claude-bot-derinworks) [reply tests/test_shared.py:188]: Out of scope for this PR. Test style refactoring can be tracked separately.

File: tests/test_shared.py [hunk 3]
```diff
@@ -138,3 +144,279 @@ def test_difficulty_prefix_length(self):
...
+                normalized_score=1.5,
+            )
+        with pytest.raises(ValidationError):
+            AIWorkEvaluation(
+                evaluation_id="eval-003",
+                submission_id="sub-001",
+                evaluator_address="a" * 64,
+                raw_score=50.0,
+                normalized_score=-0.1,
+            )
```
[the remark this answers is not in the document]
Comment (claude-bot-derinworks) [reply tests/test_shared.py:318]: Out of scope for this PR. Test style refactoring can be tracked separately.

File: shared/models/ai_work.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+class RewardDecision(BaseModel):
+    """Decision on whether and how much to reward for evaluated AI work."""
+
+    decision_id: str
+    evaluation_id: str
+    task_id: str
+    worker_address: str
+    reward_amount: float = Field(ge=0)
+    approved: bool
```
[the remark this answers is not in the document]
Comment (claude-bot-derinworks) [reply shared/models/ai_work.py:76]: Out of scope for this PR. Cross-field business logic validation can be tracked separately.

**Review by admin-derinworks — APPROVED** [review]
